src/core/server/saved_objects/migrationsv2/model/model.ts

```diff
@@ -489,12 +490,26 @@ export const model = (currentState: State, resW: ResponseType<AllActionStates>):
 
     if (Either.isRight(res)) {
       if (stateP.corruptDocumentIds.length === 0 && stateP.transformErrors.length === 0) {
-        return {
-          ...stateP,
-          controlState: 'REINDEX_SOURCE_TO_TEMP_INDEX_BULK', // handles the actual bulk indexing into temp index
-          transformedDocs: [...res.right.processedDocs],
-          progress,
-        };
+        const batches = createBatches(
+          res.right.processedDocs,
+          stateP.tempIndex,
+          stateP.batchSizeBytes
+        );
+        if (Either.isRight(batches)) {
+          return {
+            ...stateP,
+            controlState: 'REINDEX_SOURCE_TO_TEMP_INDEX_BULK', // handles the actual bulk indexing into temp index
+            transformedDocBatches: batches.right,
+            currentBatch: 0,
+            progress,
+          };
+        } else {
+          return {
+            ...stateP,
+            controlState: 'FATAL',
+            reason: `The document with _id "${batches.left.document._id}" is ${batches.left.docSizeBytes} bytes which equals or exceeds the configured maximum batch size of ${batches.left.batchSizeBytes} bytes. To proceed, please increase the migrations.batchSizeBytes Kibana configuration option and ensure that the Elasticsearch http.max_content_length configuration option is set to an equal or larger value.`,
```

Comment:
I've changed `bytes which equals or exceeds` -> `bytes which exceeds`, if the document size == maxBatchSizeBytes == http.max_content_length then the migration will succeed. We don't need to account for extra overhead since ES only checks the body, there's a different setting limiting header size.